File: Leetcode/dailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        
        int len= T.length;
        int[] fa = new int[len];
        if (len == 0)
            return fa;
        Stack<Integer> stack = new Stack<Integer>();
        stack.push (0);
        for (int i =1; i< T.length;i+=1)
        {
            while(!stack.isEmpty() && T[i] > T[stack.peek()]) 
            {
                int top = stack.pop();
                fa[top] = i - top;
            }
            stack.push (i);
        }
        while (!stack.isEmpty())
        {
            int top = stack.pop();
            fa[top] = 0;
        }
        return fa;
    }
}

File: Leetcode/Daily_Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {

        stack<int> st;
        int finalans[T.size()];
        st.push (0);
        for (int i =1 ; i< T.size();i+=1)
        {
            while  (st.size() > 0 && T[i] > T[st.top()])
            {
                int curr = st.top();
                st.pop();
                finalans [curr] = i -curr;                
            }
            st.push(i);
        }
        while (!st.empty())
        {
            int curr = st.top();
            st.pop();
            finalans[curr] = 0;
        }
    vector<int> f;
        for  (int i =0 ; i< T.size();i+=1)
            f.push_back(finalans[i]);
        return f;
    }
};
